dietpi/login

```diff
@@ -139,89 +134,92 @@
+		# Automated?
+		if grep -qi '^[[:blank:]]*AUTO_SETUP_AUTOMATED=1' /DietPi/dietpi.txt; then
 
-					G_WHIP_MSG "DietPi is currently running on another screen.\n\n$additional_text"
+			automated_setup=1
+			#	Set non-interactive shell, if automated installation (as .bashrc run via STDIN check is interactive)
+			export G_USER_INPUTS=0
 
-				else
+		fi
 
-					#	Check internet
-					optional_cmd_inputs='--no-check-certificate' G_CHECK_URL "$(grep -m1 '^[[:blank:]]*deb ' /etc/apt/sources.list | awk '{print $2}')" # Will exit on failure here then prompt user to configure network
+		# Show GPL license
+		if [[ -f /var/lib/dietpi/license.txt ]] && (( ! $automated_setup )); then
 
-					#	Check NTP synced
-					/DietPi/dietpi/func/run_ntpd
+			G_WHIP_VIEWFILE /var/lib/dietpi/license.txt
+			rm /var/lib/dietpi/license.txt
 
-					#	Show GPL license
-					if [[ -f /var/lib/dietpi/license.txt ]]; then
+		fi
 
-						G_WHIP_VIEWFILE /var/lib/dietpi/license.txt
-						rm /var/lib/dietpi/license.txt
+		until (( $G_DIETPI_INSTALL_STAGE == 2 ))
+		do
 
-					fi
+			# 1st run dietpi-update
+			if (( $G_DIETPI_INSTALL_STAGE == 0 )); then
 
-					/DietPi/dietpi/dietpi-update 1 #Sets G_DIETPI_INSTALL_STAGE=1
+				# - Prompt and exit if DietPi-Update runs in other session already
+				pgrep 'dietpi-update' &> /dev/null && Other_Session_User_Prompt
 
-				fi
+				# - Check internet
+				optional_cmd_inputs='--no-check-certificate' G_CHECK_URL "$(grep -m1 '^[[:blank:]]*deb ' /etc/apt/sources.list | mawk '{print $2}')" # Will exit on failure here then prompt user to configure network
 
-			# - 1st run dietpi-software installs
-			elif (( $G_DIETPI_INSTALL_STAGE == 1 )); then
+				# - Check NTP synced
+				/DietPi/dietpi/func/run_ntpd
 
-				#	Wait for DietPi-Software if already running, else run it
-				local run_dietpi_software=1
-				while pgrep 'dietpi-software' &> /dev/null
-				do
+				# - Start DietPi-Update
+				/DietPi/dietpi/dietpi-update 1 # Sets G_DIETPI_INSTALL_STAGE=1
 
-					run_dietpi_software=0
+				# - Update install state, to allow DietPi-Software in same loop, if no update was required
+				export G_DIETPI_INSTALL_STAGE=$(</DietPi/dietpi/.install_stage)
 
-					# - Automated
-					if (( $automated_setup )); then
+			fi
 
-						G_DIETPI-NOTIFY 2 'DietPi is currently installing and configuring your system. Please wait for this to complete, check back later.'
+			# 1st run dietpi-software installs
+			if (( $G_DIETPI_INSTALL_STAGE == 1 )); then
 
-					else
+				# - Prompt and exit if DietPi-Software runs in other session already
+				pgrep 'dietpi-software' &> /dev/null && Other_Session_User_Prompt
 
-						G_DIETPI-NOTIFY 2 'DietPi-Software is already running on another terminal/screen. Please complete or exit, before continuing.'
+				# - Start DietPi-Software
+				/DietPi/dietpi/dietpi-software | tee $FP_TMP_DIETPI_FIRSTRUNSETUP_LOG # Sets G_DIETPI_INSTALL_STAGE=2
```

Comment:
@Fourdee 
If user chooses to reboot after `dietpi-software` has finished (most do, I guess), first run log will not be saved to disk currently.
Not sure how to solve best, but writing log to disk directly at least is an easy solution.

Reply:
@MichaIng 

Yep, need to write to disk for this 👍